### ISSUE TITLE
disambiguate not working properly

### ISSUE THREAD

**Issue by srijan-mishra** [lopuhin/python-adagram#6]
Hey!

Really appreciate the hard work that you guys have done on adagram. I have trained a model on one month of reddit data and have been doing the following things through your implementation:

`vm.word_sense_probs('windows')`
gives me:
[(0, 0.026882535901865078),
 (1, 0.33745738176790857),
 (2, 0.15698251813511088),
 (3, 0.069624420466025169),
 (4, 0.40905314372909024)]

Now, for the first meaning of windows, I get the following results:
`vm.sense_neighbors('windows',0)`
gives
[(u'wn7', 0, 0.80748391),
 (u'00614', 0, 0.80702519),
 (u'oem', 0, 0.80138087),
 (u'64', 0, 0.78792632),
 (u'premium', 1, 0.78470039),
 (u'operating', 0, 0.78342617),
 (u'microsoft', 0, 0.76713639),
 (u'bit', 0, 0.76577711),
 (u'system', 1, 0.75200081),
 (u'sp1', 0, 0.7426722)]

`vm.sense_neighbors('windows',1)`
gives
[(u'7', 2, 0.91284883),
 (u'osx', 0, 0.90063703),
 (u'linux', 0, 0.88184464),
 (u'win8', 0, 0.88007188),
 (u'xp', 1, 0.87717748),
 (u'ubuntu', 0, 0.87087566),
 (u'win7', 0, 0.86827832),
 (u'x64', 0, 0.8610155),
 (u'mac', 1, 0.85330862),
 (u'os', 3, 0.84352261)]

`vm.sense_neighbors('windows',2)`
gives
[(u'doors', 1, 0.88995254),
 (u'walls', 1, 0.8533318),
 (u'floors', 0, 0.78216279),
 (u'mirrors', 0, 0.77678096),
 (u'vents', 0, 0.7755211),
 (u'doorways', 0, 0.77175504),
 (u'cabinets', 0, 0.77003133),
 (u'shutters', 0, 0.76936632),
 (u'door', 3, 0.76735002),
 (u'jamb', 0, 0.76461965)]

`vm.sense_neighbors('windows',3)`
gives
[(u'microsoft', 1, 0.88076365),
 (u'ios', 2, 0.83934981),
 (u'windows7', 0, 0.83232284),
 (u'software', 0, 0.80623454),
 (u'linux', 1, 0.78886378),
 (u'msdn', 0, 0.78856975),
 (u'browser', 1, 0.78598648),
 (u'technet', 0, 0.78191495),
 (u'macosx', 0, 0.76813924),
 (u'kernel', 1, 0.7653088)]

`vm.sense_neighbors('windows',4)`
gives
[(u'ubuntu', 0, 0.87467057),
 (u'os', 3, 0.86868304),
 (u'mac', 1, 0.86347193),
 (u'osx', 0, 0.84865129),
 (u'kernel', 0, 0.84220809),
 (u'android', 0, 0.84065717),
 (u'device', 1, 0.83707845),
 (u'software', 1, 0.83519411),
 (u'virtualbox', 0, 0.8335678),
 (u'update', 2, 0.83236873)]

Now, when I am running disambiguate I am not getting correct meaning even if I make a sentence using the top words from the meaning which corresponds to house windows. 
Example:
`vm.disambiguate('windows','doors walls floors')`
gives 
`array([  3.04453898e-09,   1.96731596e-06,   7.72059540e-14,
         9.99997322e-01,   7.08039131e-07])`

Moreover, any disabmiguate I run gives highest probablity for the 4th meaning(index 3).

Can you please suggest a fix?

**Comment by lopuhin** [comment]
Hi @srijan-mishra ! For the record, I only ported original code from Julia, all the awesome properties of AdaGram are totally not my merit :)

A quick question first - which implementation did you use for training? I'm asking because right now python-adagram is not ready for training - it's recommended that you train a model with AdaGram.jl and then convert it to python-adagram format.

**Comment by srijan-mishra** [comment]
Hey!

I trained a model with Adagram.jl and then converted it using the conversion method that you have suggested. If I used the same model to do the disambiguate task in Julia following are the results:

![image](https://cloud.githubusercontent.com/assets/20169265/25837768/f737a8b8-34ac-11e7-84c4-b09324655151.png)
![image](https://cloud.githubusercontent.com/assets/20169265/25837784/078bc442-34ad-11e7-8f5c-7c698ce78064.png)

As you can see, it is giving the correct meaning the highest probability.

I am seeing the code end to end as well. Can you too please have a look? Let's sort this issue :)

**Comment by lopuhin** [comment]
Yeah, that's very strange, looks more like a bug than anything else. FWIW, I hope it's not completely broken as I ran ``disambiguate`` on some Russian contexts and it gave sensible results :) But I would definitely like to sort that out.

**Comment by srijan-mishra** [comment]
Hey!

I have sorted this out. Have sent you a pull request. You can have a look.

Thanks
Srijan
